gno.land/pkg/gnoland/vals.go

```diff
@@ -0,0 +1,122 @@
+package gnoland
+
+import (
+	"fmt"
+	"strconv"
+
+	gnovm "github.com/gnolang/gno/gnovm/stdlibs/std"
+	abci "github.com/gnolang/gno/tm2/pkg/bft/abci/types"
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto"
+	"github.com/gnolang/gno/tm2/pkg/events"
+)
+
+const (
+	valRealm = "gno.land/r/sys/vals"
+
+	validatorAddedEvent   = "ValidatorAdded"
+	validatorRemovedEvent = "ValidatorRemoved"
+
+	addressEventKey     = "address"
+	pubKeyEventKey      = "pub_key"
+	votingPowerEventKey = "voting_power"
```

Comment:
Ideally, we should avoid using complex types in the returned value and rely on simpler types instead.

We can also explore the possibility of directly manipulating an ABCI object from the contract, thereby avoiding the need for marshalling altogether.

We can stick with this approach for now.